spec/services/kessel_rbac_spec.rb

```diff
@@ -0,0 +1,265 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe KesselRbac, type: :service do
+  let(:user) { create(:v2_user) }
+  let(:org_id) { user.org_id }
+
+  before do
+    # Mock Kessel SDK classes
+    stub_const('Kessel::Configuration', Class.new)
+    stub_const('Kessel::ApiClient', Class.new)
+    stub_const('Kessel::KesselInventoryService', Class.new)
+  end
+
+  describe '.check_permission' do
+    let(:mock_client) { double('kessel_client') }
+    let(:mock_response) { double('response', allowed: true) }
+
+    before do
+      allow(KesselRbac).to receive(:enabled?).and_return(true)
+      allow(described_class).to receive(:client).and_return(mock_client)
+    end
+
+    context 'when permission is granted' do
```

Comment:
The description of the context could say more about validity of the type.

```suggestion
    context 'when permission is granted with_invalid_identity_type' do
```